portal-impl/src/com/liferay/portlet/internal/PortletContainerImpl.java

```diff
@@ -529,8 +529,8 @@ private ActionResult _processAction(
 
 			List<Event> events = liferayActionResponse.getEvents();
 
-			String redirectLocation =
-				liferayActionResponse.getRedirectLocation();
+			String redirectLocation = PortalUtil.escapeRedirect(
```

Comment:
@jeffwu0724 shouldn't the escape be moved into the if block below?

Reply:
Hi @shuyangzhou ,

The reason why I did not move this escape into the if block is:

When following the steps in https://liferay.atlassian.net/browse/LPS-196525, `www.example.com` will be passed in `liferayActionResponse.getRedirectLocation()`, the condition `Validator.isNull(redirectLocation)` and `portlet.isActionURLRedirect()` will be False, and the escape process will never be run, and the response header will still  include the location info.

Please correct me if I made any misunderstanding. 
Thanks!

Reply:
So that means you should break the existing if conditions to make them nested.
You don't want to escape the `null` url, that won't do anything meaningful.
It will be like:
```
			if (Validator.isNull(redirectLocation)) {
				if (portlet.isActionURLRedirect()) {
					...
				}
			}
			else {
				redirectLocation = PortalUtil.escapeRedirect(redirectLocation);
			}
```